app/controllers/tasks_controller.rb

```diff
@@ -0,0 +1,68 @@
+TASKS = [
+  { id: 1, name: "Read POODR", description: "Read Ch4 and try not to cry", completion_date: "2018-09-30" },
+  { id: 2, name: "Refactor Hotel", description: "Refactor and try not to cry", completion_date: "2018-09-30"}
+]
+
+class TasksController < ApplicationController
+  def index
+    @tasks= Task.all
+  end
+
+  def show
+    id = params[:id]
+    @task = Task.find_by(id:id)
+  end
+
+  def new
+    @task = Task.new
+  end
+
+  def edit
+    id = params[:id]
+    @task = Task.find_by(id:id)
```

Comment:
What happens if the task isn't found? You should include error-handling code in all of these actions that look for a single task.